components/script/dom/testbinding.rs

```diff
@@ -493,6 +494,8 @@ impl TestBindingMethods for TestBinding {
     fn StringMozPreference(&self, pref_name: DOMString) -> DOMString {
         get_pref(pref_name.as_ref()).as_string().map(|s| DOMString::from(s)).unwrap_or_else(|| DOMString::new())
     }
+
+    fn DocumentOrTypedef(&self, _: Option<Option<DocumentOrBlobOrHTMLElementOrFormData >>) -> () {}
```

Comment:
Hmm, ok so it needs to collapse the "optional" and "?" into a single Option and keep the type hierarchy, it will need a deeper look into the codegen.